app/src/main/java/org/stepic/droid/base/FragmentActivityBase.java

```diff
@@ -103,6 +104,24 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         App.Companion.component().inject(this);
+
+        if (savedInstanceState == null && AppConstants.OPEN_NOTIFICATION.equals(getIntent().getAction())) {
+            markNotificationAsRead(getIntent());
+        }
+    }
+
+    private void markNotificationAsRead(Intent intent) {
+        final long notificationId = intent.getLongExtra(AppConstants.KEY_NOTIFICATION_ID, 0);
+        if (notificationId != 0) {
+            threadPoolExecutor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        api.setReadStatusForNotification(notificationId, true).execute();
```

Comment:
What if internet is not available? May be we should add it in some queue? See solution for viewed steps (`databaseFacade#addToQueueViewedState`). 

And using of `api` + `threadPoolExecutor` from Fragment is not good idea, because this code is not testable in unit testing. We cannot check behaviour of marking notification as read. The logic (adding in queue if internet connection is lost) will bloat this class and violate Single Responsibility principle in **S**OLID.